### DISCUSSION TITLE
Renovate doesn't update `actions/upload-artifact`

### DISCUSSION THREAD

**Discussion by palango** [renovatebot/renovate#26801] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

In https://github.com/celo-org/celo-blockchain/pull/2221 it should update all github actions. According to the logs it identifies it, but the update is not written to the PR.

### Relevant debug logs

<details><summary>Logs</summary>

```

          {
            "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
            "commitMessageTopic": "{{{depName}}} action",
            "currentValue": "v3",
            "currentVersion": "v3",
            "datasource": "github-tags",
            "depName": "actions/upload-artifact",
            "depType": "action",
            "fixedVersion": "v3",
            "isSingleVersion": true,
            "packageName": "actions/upload-artifact",
            "registryUrl": "https://github.com",
            "replaceString": "actions/upload-artifact@v3",
            "sourceUrl": "https://github.com/actions/upload-artifact",
            "versioning": "docker",
            "warnings": [],
            "updates": [
              {
                "bucket": "latest",
                "newVersion": "v4",
                "newValue": "v4",
                "releaseTimestamp": "2024-01-18T20:11:31.000Z",
                "newMajor": 4,
                "newMinor": null,
                "updateType": "major",
                "pendingChecks": true,
                "branchName": "renovate/github-actions"
              }
            ]
          },

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by lvpeschke** [comment, marked as answer]
I believe this is working as expected (or at least as configured).

The PR is based on the config in https://github.com/celo-org/celo-blockchain/blob/master/.github/renovate.json, which itself [extends](https://github.com/celo-org/celo-blockchain/blob/46de0db47a6094fcf5414901520b569d5a25c2e5/.github/renovate.json#L3-L4) https://github.com/celo-org/.github/blob/main/renovate-config.json, where we can find [`"minimumReleaseAge": "7 days"`](https://github.com/celo-org/.github/blob/eead3ad9517261d93ad0b8dcad2d8562fe097c7a/renovate-config.json#L13C4-L13C21).

The [docs](https://docs.renovatebot.com/configuration-options/#minimumreleaseage) tell us that:
> If the time since the release is less than the set minimumReleaseAge a "pending" status check is added to the branch.

This is visible in the log [above](https://github.com/renovatebot/renovate/discussions/26801#discussion-6118503): `"pendingChecks": true` and, looking at the full log, absent from `"updates"` entries that were included in the update, and the release timestamp is indeed fresher than 7 days.

**Reply by palango** [reply]
True, but there are also [older versions](https://github.com/actions/upload-artifact/releases) available for the `upload-artifact` plugin, which are older than 7 days and should be picked instead.

**Reply by rarkins** [reply]
It's a side effect do you using v3. v4 would resolve to a release younger than 7 days so is withheld.

There's no point having minimumReleaseAge for such a dependency when you have such a permission constraint

**Reply by palango** [reply]
> There's no point having minimumReleaseAge for such a dependency when you have such a permission constraint

Why not? I might want to wait before upgrading for at least a week.
At least my understanding of `minimumReleaseAge` is: "give me the highest version that's older than X days".

**Reply by rarkins** [reply]
My preference is to set major updates to on demand using dependency dashboard rather than a fixed schedule. 

Once you are on v4 you'll get whatever new v4 version is released every time you run an action.
